### PR TITLE
[PyMVA] Refactoring of sklearn classifier with multi-class capability, unit-testing and ranking

### PR DESCRIPTION
**Pull request by stwunsch**
This is a rebase and combination of the PRs #303 #314 #315.

- Refactor PyRandomForest, PyGTB, PyAdaBoost
- Add variable ranking to these classifiers
- Add unit-tests for the sklearn classifiers
- Add unit-test for PyKeras multiclass classification
- Move `GetMvaValues` implementation from base class to method to stay independent from sklearn

You can test the changes with `ctest -V -R PyMVA`.

### CODE REVIEW

File: tmva/pymva/inc/TMVA/MethodPyAdaBoost.h [hunk 1]
```diff
@@ -19,24 +19,23 @@
 //                                                                      //
 // MethodPyAdaBoost                                                     //
 //                                                                      //
-//                                                                      //
 //////////////////////////////////////////////////////////////////////////
 
+#ifndef ROOT_TMVA_PyMethodBase
```
Comment (Teemperor) [inline tmva/pymva/inc/TMVA/MethodPyAdaBoost.h]: Could you remove these redundant header guards? We just removed all of them from the current repo in #341

Comment (stwunsch) [inline tmva/pymva/inc/TMVA/MethodPyAdaBoost.h]: What is the current coding style? Moving all the guards into the headers? What about using `#pragma once`? That would be nice ;)

Comment (stwunsch) [inline tmva/pymva/inc/TMVA/MethodPyAdaBoost.h]: I'll fix it later!

Comment (Teemperor) [inline tmva/pymva/inc/TMVA/MethodPyAdaBoost.h]: Not sure about all the details of the coding style, but at least here I think it should be normal header guards in the respective header and no "double guards"around the include :)

**Comment by pcanal** [comment]
>  at least here I think it should be normal header guards in the respective header and no "double guards"around the include :)

You are correct.  We *just* changed the coding convention to not longer require the 'double guard' (which is nowadays redundant, in the past it saved in disk access).

Cheers,
Philippe.

**Comment by martinmine** [comment]
@phsft-bot build!

**Comment by martinmine** [comment]
@phsft-bot build!

**Comment by martinmine** [comment]
@phsft-bot build!

**Comment by martinmine** [comment]
@phsft-bot build!

**Comment by martinmine** [comment]
@phsft-bot build!

**Comment by lmoneta** [comment]
Hi Stefan, 

I would like to merge this PR. Can you please verify the shown conflict with the master branch ? 
Thank you 
 Lorenzo

**Comment by stwunsch** [comment]
I've rebased and solved the conflict!

**Comment by lmoneta** [comment]
Thank you. It is now merged

**Pull request merged**